Fix for stats endpoints being called twice

ravishanker opened the pull request:
This PR replaces ViewPager with ViewPager2.  Legacy ViewPager is known to be buggy.  Infact, part of the problem seems to be due to onTabSelected invoked inconsistently, on same fragment instantiation that was causing duplicate calls.

Fixes #15906 

To test:

- Setup a debugging proxy like [Charles](https://www.charlesproxy.com/) or [Proxyman](https://proxyman.io/) with your device or emulator as you need to observe network traffic
- Install the app
- Launch app, on My Site, click on Stats from quick links or Stats option below
- Ensure, Stats launches in Insights tab
- Observe network calls and make sure there are no repeated calls to the same endpoint
- Switch to Day, Week, Month and Year tabs and observe network calls.  Ensure there are no repeated calls.

NOTE: By default the ViewPager instantiates next tab fragment offscreen.  So for e.g. when on Insights tab, it instantiates Day fragment and when on Day tab, then Week fragment, so on.  That is to say, You'll see current tab calls plus next tab calls

## Regression Notes
1. Potential unintended areas of impact
Tested all stats tabs variations

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and existing unit tests

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Replace buggy ViewPager with ViewPager2

Part of the problem seems to be due to onTabSelected invoked inconsistently, on same fragment instantiation

ravishanker committed:
Fix ciktlint indentation issue

renanferrari reviewed (commented):
Ok, so this was actually kind of hard for me to test, not because of the changes in this PR specifically, but because the issue itself was hard to reproduce. 

I have two testing devices, a Motorola One and a Pixel 3 XL, and I could only reproduce the issue on the Motorola, but I can't figure out why. The PR changes look good to me and they worked fine on the Pixel 3XL but then on the Motorola, [I got an ANR](https://user-images.githubusercontent.com/830056/153678765-a492d66c-36ff-4122-be8e-9f2276382df4.png):

https://user-images.githubusercontent.com/830056/153677361-e7217d44-b588-4ffa-a305-9e84eb71b20e.mp4

Either way, I think it's a good idea to replace `ViewPager` with `ViewPager2` even if it doesn't solve the duplicate request issues (unless, of course, we keep seeing this ANR).

ravishanker commented:
> then on the Motorola, I got an ANR:

@renanferrari, Thanks for discovering the ANR.  I found the issue, after much tapping around.  

```
2022-02-14 10:46:42.359 10453-10453/org.wordpress.android.prealpha E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.wordpress.android.prealpha, PID: 10453 java.lang.IllegalStateException: Page(s) contain a 
ViewGroup with a LayoutTransition (or animateLayoutChanges="true"), which interferes with the scrolling 
animation. Make sure to call getLayoutTransition().setAnimateParentHierarchy(false) on all ViewGroups with 
a LayoutTransition before an animation is started.
        at androidx.viewpager2.widget.ScrollEventAdapter.updateScrollEventValues(ScrollEventAdapter.java:272)
        at androidx.viewpager2.widget.ScrollEventAdapter.onScrolled(ScrollEventAdapter.java:178)
        at androidx.recyclerview.widget.RecyclerView.dispatchOnScrolled(RecyclerView.java:5173)
        at androidx.recyclerview.widget.RecyclerView$ViewFlinger.run(RecyclerView.java:5338)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
        at android.view.Choreographer.doCallbacks(Choreographer.java:845)
        at android.view.Choreographer.doFrame(Choreographer.java:775)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7839)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

Looks like it is a [well known issue](https://android-developers.googleblog.com/2011/11/android-40-graphics-and-animations.html) that happens.  Luckily the error message has solution to the problem as well.  I pushed a commit with the fix by adding getLayoutTransition().setAnimateParentHierarchy(false).  Another way to fix this could be removing layoutTransition on the layout and [adding layoutTransition.enableTransitionType(LayoutTransition.CHANGING)](https://www.youtube.com/watch?v=55wLsaWpQ4g)

Please test it again.  Thank you.

ravishanker committed:
Fix for ANR due to layout scrolling animation

Fix for Page(s) contain a
ViewGroup with a LayoutTransition (or animateLayoutChanges="true"), which interferes with the scrolling
animation.

Adding getLayoutTransition().setAnimateParentHierarchy(false) on all ViewGroups with
a LayoutTransition before an animation is started

renanferrari reviewed (approved):
>I pushed a commit with the fix by adding getLayoutTransition().setAnimateParentHierarchy(false)

Good finding! I can confirm I don't see the ANR anymore. That said, I'm still seeing duplicated requests on one of my devices 😕

I'm going to look into this and see what I can find. Either way, as I mentioned before, I think we should go ahead and merge this since it introduces some useful changes.
